mono/metadata/threads-types.h

```diff
@@ -289,7 +289,7 @@ gboolean
 mono_thread_create_checked (MonoDomain *domain, gpointer func, gpointer arg, MonoError *error);
 
 /* Can't include utils/mono-threads.h because of the THREAD_INFO_TYPE wizardry */
-void mono_threads_add_joinable_runtime_thread (gpointer thread_info);
+void mono_threads_add_joinable_runtime_thread (MonoThreadInfo *thread_info);
```

Comment:
comment above this line about `THREAD_INFO_TYPE` seems relevant - this should probably be `struct _MonoThreadInfo*` (because `MonoThreadInfo` typedef is not necessarily visible every time this header is included).

Reply:
Yeah I was kinda expecting it not to compile but it did.

Reply:
Yea we really need a fragile header CI lane.  

`gcc -Wall -Werror -DHEADER=threads-types.h -c wrapper.c`
wrapper.c:
```c
#include "config.h"
#include HEADER
/* include twice to verify that there are include guards */
#include HEADER
int main () { return 0; }
```

Reply:
I think the code is ok and the comment is wrong. I will remove the comment.
The header is included and does have an include guard, and the typedef is always visible.

The header does have varying definitions, gross, but I don't see how it affects this line.

This might  be somewhat or completely improvable with C++ with overloads or derivatin (SgenThreadInfo from ThreadInfo), at least for input parameters, not return values -- there are some of each.